### PR TITLE
Implement notification context and request processing engine

### PR DESCRIPTION
**Pull request by Crispried**
Notification context allows to wrap the app component in NotificationContextProvider and then access addNotification function through the NotificationContext. So, it makes notifications global and it won't be necessary to care anymore about component for notification. All what you need to do is just add new notification from anywhere and it will work!

RequestProcessingHelper exposes computeFinalRequestProcessingState which computes the final states depends on passed states. Using this function it will be possible to determine the global state of your "business action" which could contain multiple "http requests". 

It will be helpful also in context of **chain of request** which highly likely will be built in the future.

### CODE REVIEW

File: src/components/MaterialUI/Feedback/NotificationContext.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+export const NotificationContext = createContext();
+
+export const NotificationContextProvider = ({ children }) => {
+	const [snackPack, setSnackPack] = useState([]);
+
+	const addNotification = useNotification(snackPack, setSnackPack);
+
+	return (
+		<NotificationContext.Provider value={{ addNotification }}>
```
Comment (TONYFECTEAU) [inline src/components/MaterialUI/Feedback/NotificationContext.js:13]: Just curious, why the "."?

File: src/utils/requestProcessingHelper.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+				return requestStates.fail;
+			case requestStates.processing:
+				return requestStates.processing;
+			case requestStates.idle:
+				idleCount++;
+				continue;
+			case requestStates.success:
+				continue;
+			default:
+				throw Error("You have to pass just requestStates type");
```
Comment (TONYFECTEAU) [inline src/utils/requestProcessingHelper.js:19]: Only valid requestStates can be used

File: src/components/MaterialUI/Feedback/NotificationContext.js (hunk 1 above)
Comment (Crispried) [reply src/components/MaterialUI/Feedback/NotificationContext.js:13]: createContext creates object with 2 main properties - Provider and Consumer. Depends on your need you should use one of those 2

File: src/utils/requestProcessingHelper.js (hunk 2 above)
Comment (Crispried) [reply src/utils/requestProcessingHelper.js:19]: Is there really big difference between those 2 statements? 😄

**Pull request merged**